pkg/apiserver/registry/datapackaging/package_storage_client.go

```diff
@@ -152,7 +152,28 @@ func (t PackageTranslator) ToExternalWatcher(intObjWatcher watch.Interface, fiel
 }
 
 func (t PackageTranslator) ToExternalError(err error) error {
-	// TODO: implement
+	statusErr, isStatusError := err.(*errors.StatusError)
+	if !isStatusError {
+		return err
+	}
+
+	details := statusErr.ErrStatus.Details
+	if details.Kind == "internalpackages" && details.Group == internalpkgingv1alpha1.SchemeGroupVersion.Group {
```

Comment:
Let's also do a nil check for details. Also, we can just return if it one of these conditions is not true.